packages/core/src/stryker.ts

```diff
@@ -45,8 +45,10 @@ export class Stryker {
         return mutantResults;
       } catch (error) {
         const log = loggerProvider.resolve(commonTokens.getLogger)(Stryker.name);
-        log.debug('Not removing the temp dir because an error occurred');
-        mutantInstrumenterInjector.resolve(coreTokens.temporaryDirectory).removeDuringDisposal = false;
+        if (mutantInstrumenterInjector.resolve(coreTokens.temporaryDirectory).removeDuringDisposal !== 'always') {
```

Comment:
```suggestion
        if (mutantInstrumenterInjector.resolve(commonTokens.options).cleanTempDir !== 'always') {
```

You should be able to use the `commonTokens` here to let the `StrykerOptions` be injected. You can use that to set the `removeDuringDisposal` to either `true` or `false`.